DependencyInjectionContainer/ServiceInstancesList.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace DependencyInjectionContainer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+internal sealed class ServicesInstanceList : IDisposable
+{
+    private List<object> instances = new List<object>();
+
+    public void Dispose()
+    {
+        for(int i = instances.Count - 1; i >= 0; i--)
+        {
+            if(instances[i] is IDisposable)
+            {
+                (instances[i] as IDisposable).Dispose();
+            }
+        }
+        instances.Clear();
+    }
+
+    internal void Add(object instance)
+    {
+        if (instances.All(item => !ReferenceEquals(item, instance)))
```

Comment:
List is too slow for this operation, consider something faster which fulfill the purpose